diff --git a/src/freedreno/vulkan/tu_device.c b/src/freedreno/vulkan/tu_device.c
--- a/src/freedreno/vulkan/tu_device.c
+++ b/src/freedreno/vulkan/tu_device.c
@@ -234,6 +234,18 @@
    instance->physical_device_count = -1;
    instance->api_version = TU_API_VERSION;
+
+   if (!getenv("FD_DEV_FEATURES")) {
+       setenv("FD_DEV_FEATURES", "enable_tp_ubwc_flag_hint=1", 1);
+   }
+   if (!getenv("MESA_SHADER_CACHE_MAX_SIZE")) {
+       setenv("MESA_SHADER_CACHE_MAX_SIZE", "1024M", 1);
+   }
+   if (!getenv("TU_DEBUG")) {
+       setenv("TU_DEBUG", "force_unaligned_device_local", 1);
+   }
